pkg/metrics/metrics.go

```diff
@@ -71,71 +68,46 @@ func NewPrometheusNodeConfig(clientset *kubernetes.Clientset) (*PrometheusNodeCo
-func Add(ctx context.Context, cfg *rest.Config, namespace string) error {
-	// Add to the below struct any other metrics ports you want to expose.
-	servicePorts := []v1.ServicePort{
-		{Port: Port, Name: PortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: Port}},
-	}
-
-	// Create Service object to expose the metrics port(s).
-	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+// Validate will verify if the Services and Service Monitors are created in the operator namespace
+func Validate(ctx context.Context, cfg *rest.Config, namespace string) error {
+	oclient, err := k8sclient.NewForConfig(cfg)
 	if err != nil {
-		return errors.Wrap(err, "could not create metrics Service")
+		return errors.Wrap(err, "could not create config clientset")
 	}
 
-	// the name for the metrics resources is set during creation of metrics service and is equivalent to the service name
-	windowsMetricsResource = service.GetName()
-
-	// Create a monitoring client to interact with the ServiceMonitor object
-	mclient, err := monclient.NewForConfig(cfg)
+	// Validate if metrics label is added to namespace
+	wincNamespace, err := oclient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
 	if err != nil {
-		return errors.Wrap(err, "could not create monitoring client")
+		return errors.Wrap(err, "could not get operator namespace")
 	}
-
-	// In the case of an operator restart, a previous SM object will be deleted and a new one will
-	// be created. We are deleting to ensure that the SM always exists with the correct spec. Otherwise,
-	// metrics may exhibit unexpected behavior if created by a previous version of WMCO.
-	err = mclient.ServiceMonitors(namespace).Delete(context.TODO(), windowsMetricsResource, metav1.DeleteOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
-		return errors.Wrap(err, "could not delete existing ServiceMonitor object")
+	if wincNamespace.Labels["openshift.io/cluster-monitoring"] != "true" {
+		metricsEnabled = false
 	}
 
-	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
-	// necessary to configure Prometheus to scrape metrics from this operator.
-	services := []*v1.Service{service}
-	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
-	if err != nil {
-		log.Error(err, "could not create ServiceMonitor object")
-		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
-		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
-		if err == metrics.ErrServiceMonitorNotPresent {
-			metricsEnabled = false
-			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
-
-		}
+	// Check if metrics service exists
+	serviceList, err := oclient.CoreV1().Services(namespace).List(ctx,
+		metav1.ListOptions{LabelSelector: "name=windows-machine-config-operator"})
```

Comment:
Where is `windows-machine-config-operator` coming from?
Instead of using a string here lets use a variable/constant if possible.